uui-core/src/helpers/applyColumnsConfig.ts

```diff
@@ -28,11 +28,12 @@ export const getColumnsConfig = <TItem, TId>(columns: DataColumnProps<TItem, TId
 
     columns.forEach(column => {
         const order = getOrderBetween(prevOrder, null);
-        const columnConfig = config[column.key] || {};
+        const hasPrevColumnConfig = !!config[column.key];
+        const columnConfig = hasPrevColumnConfig ? config[column.key] : {};
 
         config[column.key] = {
             width: columnConfig.width || column.width,
-            fix: columnConfig.fix || column.fix,
+            fix: hasPrevColumnConfig ? columnConfig.fix : column.fix,
```

Comment:
May be we can simplify it to `config[column.key]?.fix || column.fix`?

Reply:
We can't. If `config[column.key].fix` will be undefined, it will be rewritten by `column.fix`, which is the source of the existing bug.